### PR TITLE
Cluster test cases

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
Adding test cases to start, kill, shutdown and restart functionalities for MB cluster

## Goals
Achieving the test cases to be run perfectly in MB cluster 

## Test environment
JDK version - 1.8.0_172
Mac OS X 10.13.5

### CODE REVIEW

File: modules/integration/src/test/java/io/ballerina/messaging/broker/integration/cluster/KillTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    @Test(dependsOnMethods = "testKillNodeOne")
+    public void testNodeTwoActive(String portTwo, String hostnameTwo) throws NamingException, JMSException {
+        Awaitility.await().atMost(30, TimeUnit.SECONDS)
+                .pollInterval(3, TimeUnit.SECONDS)
+                .until(() -> !ClusterUtils.isPortAvailable(hostnameTwo, Integer.parseInt(portTwo)));
+        Assert.assertFalse(ClusterUtils.isPortAvailable(hostnameTwo, Integer.parseInt(portTwo)),
+                "Broker node two is not active");
+    }
+
+    @Test(dependsOnMethods = "testNodeTwoActive")
```
Comment (malakaganga) [inline modules/integration/src/test/java/io/ballerina/messaging/broker/integration/cluster/KillTest.java]: Shall we add test method description here?

Comment (praveennadarajah) [reply modules/integration/src/test/java/io/ballerina/messaging/broker/integration/cluster/KillTest.java]: Fixed. Added method descriptions

File: modules/integration/src/test/resources/cluster-suite.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        </groups>
+        <classes>
+            <class name="io.ballerina.messaging.broker.integration.cluster.KillTest" />
+            <class name="io.ballerina.messaging.broker.integration.cluster.StartTest" />
+            <class name="io.ballerina.messaging.broker.integration.cluster.ShutdownTest" />
+            <class name="io.ballerina.messaging.broker.integration.cluster.RestartTest" />
+        </classes>
+    </test>
+
+</suite>
```
Comment (isudana) [inline modules/integration/src/test/resources/cluster-suite.xml]: New line at the end missing

File: modules/integration/src/test/java/io/ballerina/messaging/broker/integration/util/ClusterUtils.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            throws IOException {
+        shutdownBrokerNode(portOne);
+        Awaitility.await().atMost(30, TimeUnit.SECONDS)
+                .pollInterval(2, TimeUnit.SECONDS)
+                .until(() -> !ClusterUtils.isPortAvailable(hostnameTwo, Integer.parseInt(portTwo)));
+        startBrokerNode(brokerHome);
+    }
+
+    /**
+     * Method for check port availability
```
Comment (isudana) [inline modules/integration/src/test/java/io/ballerina/messaging/broker/integration/util/ClusterUtils.java]: Following is a good read on how to write javadoc properly.
https://www.oracle.com/technetwork/articles/java/index-137868.html

File: modules/integration/src/test/resources/cluster-suite.xml (hunk 2 above)
Comment (praveennadarajah) [reply modules/integration/src/test/resources/cluster-suite.xml]: Fixed

File: modules/integration/src/test/java/io/ballerina/messaging/broker/integration/util/ClusterUtils.java (hunk 3 above)
Comment (praveennadarajah) [reply modules/integration/src/test/java/io/ballerina/messaging/broker/integration/util/ClusterUtils.java]: I will follow and update

**Pull request closed without merging**